packages/logs/src/domain/contexts/trackingConsentContext.ts

```diff
@@ -0,0 +1,15 @@
+import { DISCARDED, HookNames } from '@datadog/browser-core'
+import type { TrackingConsentState } from '@datadog/browser-core'
+import type { Hooks } from '../hooks'
+
+export function startTrackingConsentContext(hooks: Hooks, trackingConsentState: TrackingConsentState) {
+  hooks.register(HookNames.Assemble, ({ startTime }) => {
+    const wasConsented = trackingConsentState.isGranted(startTime)
+
+    if (!wasConsented) {
+      return DISCARDED
+    }
+
+    return SKIPPED
```

Comment:
💬 suggestion : it should probably return an empty object to match the test, or otherwise the test need to match SKIPPED
```suggestion
    return {}
```

Reply:
I'll fix the test. I accepted the change per @amortemousque recommendation